arches_component_lab/views/api/relatable_resources.py

```diff
@@ -34,44 +34,35 @@ def get(self, request, graph, node_alias):
             "name", "graphid"
         )
 
-        resources = ResourceInstance.objects.filter(graph_id__in=graphs).annotate(
-            order_field=Case(
-                When(resourceinstanceid__in=initial_values, then=Value(0)),
-                default=Value(1),
-            )
+        resources = ResourceInstance.objects.filter(graph_id__in=graphs).exclude(
+            resourceinstanceid__in=initial_values
         )
 
-        query_string = "descriptors__{}__name__icontains".format(language)
-
         if filter_term:
             resources = resources.filter(
-                Q(**{query_string: filter_term})
-                | Q(resourceinstanceid__in=initial_values)
+                 Q(**{"descriptors__{}__name__icontains".format(language): filter_term})
             )
 
-        paginator = Paginator(
-            resources.order_by(
-                "order_field", "descriptors__{}__name".format(get_language())
-            ),
-            items_per_page,
-        )
-        page_object = paginator.get_page(page_number)
+        offset = (int(page_number) - 1) * int(items_per_page)
+        limit = (int(offset)) + int(items_per_page)
+
         data = [
             {
                 "resourceinstanceid": resource.resourceinstanceid,
                 "display_value": resource.descriptors[language]["name"],
-                "order_field": resource.order_field,
             }
-            for resource in page_object
+            for resource in resources.order_by(
+                "descriptors__{}__name".format(language)
+            )[offset:limit]
```

Comment:
For low values of `items_per_page` I don't think it would matter, but in general if large sets can come back and we're throwing away the model instances, we can use `.values()` to get back dicts, e.g. add `.values("resourceinstanceid", "name")` and change the dot-access to `["resourceinstanceid"]` and `["name"]`.